router/coursecategoryrouter.js

```diff
@@ -0,0 +1,140 @@
+var express = require("express");
+require("express-async-errors");
+var router = express.Router();
+
+const {body, query, param} = require("express-validator");
+const { commonValidate } = require("../middleware/expressValidator");
+
+var courseCategoryController = require("../controller/coursecategorycontroller");
+
+/**
+ * @openapi
+ * '/api/courseCategory/addCourseCategory':
+ *  post:
+ *     tags:
+ *     - CourseCategory Controller
+ *     summary: add course category
+ *     description: add course category
+ *     security:
+ *       - BearerAuth: []
+ *     requestBody:
+ *      required: true
+ *      content:
+ *        application/json:
+ *           schema:
+ *            type: object
+ *            required:
+ *              - categoryname
+ *              - categorylevel
+ *              - categoryparentid
+ *            properties:
+ *              categoryname:
+ *                type: string
+ *                default: Node
+ *              categorylevel:
+ *                type: number
+ *                default: 5
+ *              categoryparentid:
+ *                type: number
+ *                default: 1
+ *     responses:
+ *      201:
+ *        description: Created
+ *      400:
+ *        description: Bad Request
+ *      404:
+ *        description: Not Found
+ *      409:
+ *        description: Conflict
+ *      500:
+ *        description: Server Error
+ */
+router.post(
+    "/addCourseCategory",
```

Comment:
you already have the path /api/courseCategory, maybe consider remove addCourseCategory and let the HTTP method POST itself convey the action.